ctapipe/reco/hillas_intersection.py

```diff
@@ -191,24 +186,15 @@ def _predict(self, hillas_dict, array_pointing, telescopes_pointings=None):
         hillas_dict_mod = {}
 
         for tel_id, hillas in hillas_dict.items():
-            if isinstance(hillas, CameraHillasParametersContainer):
-                focal_length = self.subarray.tel[tel_id].optics.equivalent_focal_length
-                camera_frame = CameraFrame(
-                    telescope_pointing=telescopes_pointings[tel_id],
-                    focal_length=focal_length,
-                )
-                cog_coords = SkyCoord(x=hillas.x, y=hillas.y, frame=camera_frame)
-                cog_coords_nom = cog_coords.transform_to(nom_frame)
-            else:
-                telescope_frame = TelescopeFrame(
-                    telescope_pointing=telescopes_pointings[tel_id]
-                )
-                cog_coords = SkyCoord(
-                    fov_lon=hillas.fov_lon,
-                    fov_lat=hillas.fov_lat,
-                    frame=telescope_frame,
-                )
-                cog_coords_nom = cog_coords.transform_to(nom_frame)
+            telescope_frame = TelescopeFrame(
+                telescope_pointing=telescopes_pointings[tel_id]
+            )
+            cog_coords = SkyCoord(
```

Comment:
Re our discussion about SkyCoord vs. Frames: could you also check here if doing `TelescopeFrame(fov_lon, ...)` is faster?

Reply:
Yes it's faster. For 200 events:
With SkyCoord: `1.06 ± 0.01`
With Frame: `0.86±0.01`